addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -76,66 +76,273 @@ function factory(dependencies) {
                     // uuid notification, only for (livechat) public handler
                     return;
                 }
-                if (typeof message === 'object') {
-                    switch (message.type) {
-                        case 'mail.channel_joined':
-                            return this._handleNotificationChannelJoined(message.payload);
-                    }
-                }
-                const [, model, id] = channel;
-                switch (model) {
-                    case 'ir.needaction':
-                        return this._handleNotificationNeedaction(message);
-                    case 'mail.channel':
-                        return this._handleNotificationChannel(id, message);
-                    case 'res.partner':
+                const [, , id] = channel;
+                switch (message.type) {
+                    /**
+                     * This notification trigger when new message are read inside a channel.
+                     * This is only send to the current user for multi-device synchronisation.
```

Comment:
Several typo/grammar issue in all of those comments. Also we should probably move them into the docstring of the method itself instead (the end goal is to get rid of this switch and directly bind handlers to type).